types/src/validator_verifier.rs

```diff
@@ -268,6 +261,17 @@ impl ValidatorVerifier {
     }
 }
 
+/// Returns sum of voting power from Map of validator account addresses, validator consensus info
+pub fn sum_voting_power(
+    address_to_validator_info: &BTreeMap<AccountAddress, ValidatorConsensusInfo>,
+) -> u64 {
+    address_to_validator_info.values().fold(0, |sum, x| {
+        // The voting power of any node is assumed to be small relative to u64::max_value()
+        assume!(sum <= u64::max_value() - x.voting_power);
```

Comment:
cc @hermanventer 

this assume is not great imo, I think it's fine to crash if we do overwrap, I would change the addition instead here:

```rust
sum.checked_add(x.voting_power).expect("sum of all voting power is greater than u64::max")
```

Reply:
It is only appropriate to make an assumption when there are compelling (and documented) reasons to believe that the assumption can never be violated. In this case, the caller can supply an argument that might overflow. Hopefully no current caller does so, but a future change might result in such a caller, so a precondition would be better than an assumption.

MIRAI has some heuristics that matter here: If you unwrap and Option with expect, it acts like an assumption. If you use unwrap, it becomes a proof obligation and MIRAI should issue a diagnostic if it cannot prove that the unwrap will never fail.

Reply:
It is a heuristic to reduce false positives. In this case it is probably better to use unwrap_or_else(|| unrecoverable!("...")).

Reply:
it's starting to be a mouthful :P I'm also not sure I see the point of unrecoverable! since panic, unwrap, except are all functions aimed at handling unrecoverable errors.

Reply:
You wan't to make it clear that an error is expected but unrecoverable, hence the advice to use unrecoverable! rather than just panic!. Looking at this again, it is perhaps a bad example of a non-recoverable error. This is much more of precondition and when it fails it is a programming mistake.